website/docs/plugin/framework/deprecations.mdx

```diff
@@ -0,0 +1,1282 @@
+1. Ensure the changelog has an entry noting the removal.
+1. Release the `MAJOR` version.
+1. In the next `MAJOR` version, remove all code associated with the removed data source or resource. Remove the resource or data source documentation page.
+1. Release the `MAJOR` version.
+
+### Example Resource Removal
+
+Given this sample provider and resource:
+
+```go
+// ... provider implementation ...
+
+func (p *exampleProvider) Resources(ctx context.Context) []func() resource.Resource {
+	return []func() resource.Resource{
+		//... other resources ...
+		NewWidgetResource,
+	}
+}
+```
+
+```go
+func NewWidgetResource() resource.Resource {
+	return &exampleWidgetResource{}
+}
+
+func (e *exampleWidgetResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// ... other configuration ...
+
+		Attributes: map[string]schema.Attribute{
+			// ... other attributes ...
+		},
+	}
+}
+
+// ... resource implementation ...
+```
+
+In order to deprecate `example_widget`, this sample can be written as:
+
+```go
+// ... provider implementation ...
+
+func (p *exampleProvider) Resources(ctx context.Context) []func() resource.Resource {
+	return []func() resource.Resource{
+		//... other resources ...
+		NewWidgetResource,
+	}
+}
+```
+
+```go
+func NewWidgetResource() resource.Resource {
+	return &exampleWidgetResource{}
+}
+
+func (e *exampleWidgetResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// ... other configuration ...
+
+		Attributes: map[string]schema.Attribute{
+			// ... other attributes ...
+		},
+		DeprecationMessage: "use example_thing resource instead",
+	}
+}
+
+// ... resource implementation ...
+```
+
+To soft remove `example_widget` with a friendly error message, this sample can be written as:
+
+```go
+// ... provider implementation ...
+
+func (p *exampleProvider) Resources(ctx context.Context) []func() resource.Resource {
+	return []func() resource.Resource{
+		//... other resources ...
+		NewWidgetResource,
+	}
+}
+```
+
+```go
+func NewWidgetResource() resource.Resource {
+	return &exampleWidgetResource{}
+}
+
+func (e *exampleWidgetResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// ... other configuration ...
+
+		Attributes: map[string]schema.Attribute{
+			// ... other attributes ...
+		},
+		DeprecationMessage: "use example_thing resource instead",
+	}
+}
+
+func (e *exampleWidgetResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
```

Comment:
Same as above, I transcribed the original SDK deprecation code examples over, but is there a better place to return this `diag` when deprecating an entire resource?

Original example: https://developer.hashicorp.com/terraform/plugin/sdkv2/best-practices/deprecations#provider-data-source-or-resource-removal